### PR TITLE
Show featured assets first

### LINKED ISSUE #270: Asset List: show featured assets at the top
In the main asset list and in the send asset list we should always have L-BTC, USDt, LCAD in the top position, the other does not matter 

image

### PR DESCRIPTION
**Pull request by bordalix**
Closes #270 

Sorts list of assets by the following criteria:
- first, if present, a new asset (used in receive list), ticker = Any
- afterwards, if present, featured assets order by L-BTC, USDT and LCAD
- finally, all remaining assets in no particular order

Sorts in the following screens:
- home
- receive (with new asset on top)
- send

@tiero please review

### CODE REVIEW

File: src/presentation/utils/sort.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * - first, if present, a new asset (used in receive list), ticker = Any
+ * - afterwards, if present, featured assets order by L-BTC, USDT and LCAD
+ * - finally, all remaining assets in no particular order
+ * @param assets list of assets in no particular order
+ * @returns assets sorted by criteria defined above
+ */
+export function sortAssets(
+  assets: (Asset & { assetHash: string })[]
+): (Asset & { assetHash: string })[] {
+  const sortedFeaturedTickers = ['Any', 'L-BTC', 'USDT', 'LCAD'];
```
Comment (tiero) [inline src/presentation/utils/sort.ts:14]: I personally think should always be L-BTC first and Any as last

Comment (tiero) [reply src/presentation/utils/sort.ts:14]:
```suggestion
  const sortedFeaturedTickers = ['L-BTC', 'USDT', 'LCAD', 'Any'];
```

Comment (tiero) [reply src/presentation/utils/sort.ts:14]:
```suggestion
  const sortedFeaturedTickers = ['L-BTC', 'USDT', 'LCAD'];
```

File: src/presentation/utils/sort.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  const featuredAssets: (Asset & { assetHash: string })[] = [];
+  for (const ticker of sortedFeaturedTickers) {
+    for (const asset of assets) {
+      if (ticker === asset.ticker) {
+        featuredAssets.push(asset);
+      }
+    }
+  }
+  const remainingAssets = assets.filter((asset) => !sortedFeaturedTickers.includes(asset.ticker));
+  return [...featuredAssets, ...remainingAssets];
```
Comment (tiero) [inline src/presentation/utils/sort.ts:24]:
```suggestion
  return [...featuredAssets, ...remainingAssets, 'Any'];
```

File: src/presentation/utils/sort.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { Asset } from '../../domain/assets';
+
+/**
+ * Takes a list of assets, and sort it by the following criteria:
+ * - first, if present, a new asset (used in receive list), ticker = Any
```
Comment (tiero) [inline src/presentation/utils/sort.ts:5]:
```suggestion

```

File: src/presentation/utils/sort.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+ * - afterwards, if present, featured assets order by L-BTC, USDT and LCAD
```
Comment (tiero) [inline src/presentation/utils/sort.ts:6]:
```suggestion
 * - first, featured assets order by L-BTC, USDT and LCAD
```

File: src/presentation/utils/sort.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+ * - finally, all remaining assets in no particular order
```
Comment (tiero) [inline src/presentation/utils/sort.ts:7]:
```suggestion
 * - all remaining assets in no particular order
```

File: src/presentation/utils/sort.ts (hunk 2 above)
Comment (bordalix) [reply src/presentation/utils/sort.ts:24]: Will not work with the change made above (removing 'Any' from featured lists).
I will change it in order to present Any always at the end.

Comment (bordalix) [reply src/presentation/utils/sort.ts]: Change commited

File: src/presentation/components/asset-list-screen.tsx [hunk 6]
```diff
@@ -27,16 +28,18 @@ const AssetListScreen: React.FC<AssetListProps> = ({
 }) => {
   const history = useHistory();
 
+  const sortedAssets = sortAssets(assets);
```
[the remark this answers is not in the document]
Comment (tiero) [reply src/presentation/components/asset-list-screen.tsx:31]: makes sense, this will not update after first mount

**Pull request merged**